subprojects/docs/src/docs/userguide/jvm/java_plugin.adoc

```diff
@@ -515,6 +515,20 @@ To help you understand how incremental compilation works, the following provides
 * Having a source structure that does not match the package names, while legal for compilation, might end up causing trouble in the toolchain.
     Even more if annotation processing and <<build_cache.adoc#build_cache,caching>> are involved.
 
+Additionally, Gradle might temporary change the output location while running incremental compilation.
+This might affect some annotation processors that inspect output locations and which allow users to wire some action to file paths (e.g. `-XepExcludedPaths` in Error Prone).
```

Comment:
```suggestion
This might affect some annotation processors that inspect output locations or accept file paths as arguments (e.g. `-XepExcludedPaths` in Error Prone).
```